code/modules/power/lighting.dm

```diff
@@ -102,10 +120,34 @@
 
 	newlight.setDir(dir)
 	transfer_fingerprints_to(newlight)
+	if(cell)
+		newlight.cell = cell
+		cell.forceMove(newlight)
+		cell = null
 	qdel(src)
 
 /obj/machinery/light_construct/attackby(obj/item/W, mob/living/user, params)
 	add_fingerprint(user)
+	if(istype(W, /obj/item/stock_parts/cell))
+		if(!cell_connectors)
+			to_chat(user, "<span class='warning'>This [src] can't support a power cell!</span>")
```

Comment:
```suggestion
			to_chat(user, "<span class='warning'>This [name] can't support a power cell!</span>")
```

Will otherwise result in `This the light fixture frame`